Add a create a version branch doc (#275)
PrestaShop/open-source: content/maintainers-guide/processes/release/prestashop/create-version-branch.md

---
title: Create a new version branch
weight: 20
aliases:
  - /maintainers-guide/releasing-prestashop/create-version-branch/
---

# Create a new branch for a new major/minor version.

## 1) Create the new branch and push it to the main repo

> ## 2) Open PRs based on the new branch and modify those files :

Reviewer: ```suggestion
## 2) Open PRs based on the new branch and modify those files:
```

> There are multiples repo to update.

Reviewer: ```suggestion
To create the new branch, we need to update several repositories to complete the necessary work.
```

> We'll list the file we modified for the creation of the 9.0.x branch.

Reviewer: ```suggestion
Below is a list of repositories with modified files for creating the `9.0.x` branch.
```

> 
> Changed files :

Reviewer: ```suggestion
```

> PrestaShop/PrestaShop :

Reviewer: ```suggestion
`PrestaShop/PrestaShop`:
```

- .github/workflows/cron_js_routing.yml
- .github/workflows/cron_nightly_build.yml
- .github/workflows/cron_nightly_tests_9.0.x_mariadb.yml ( created )
- .github/workflows/cron_nightly_tests_9.0.x_mysql.yml ( created )
- .github/workflows/cron_nightly_tests_reports.yml
- .github/workflows/cron_nightly_tests_reusable.yml
- .github/workflows/cron_php_update_modules.yml
- .github/PULL_REQUEST_TEMPLATE.md

> PrestaShop/test-scenarios :

Reviewer: ```suggestion
`PrestaShop/test-scenarios`:
```

- .github/workflows/gh-pages.yml
- config.json

> PrestaShop/presthubot :

Reviewer: ```suggestion
`PrestaShop/presthubot`:
```

- src/App/Command/SlackNotifierCommand.php

> PrestaShop/prestashop.github.io : ( no PR needed this time )

Reviewer: ```suggestion
`PrestaShop/prestashop.github.io`: (you can commit the changes directly, no need to create a pull request)
```

Author: That's not what I meant. I meant that there was on changes needed on this repo. They had already been done.

Reviewer: I'm not sure I understand, then, from what you're writing, this information is irrelevant. The change wasn't required this time, but it might be in the future, right? Do I understand correctly? Remember that you don't document what happened in v9, you document something for the future use 👍🏻

> - data/tags.yml ( check if already updated )

Reviewer: ```suggestion
- data/tags.yml (make sure the tag doesn't exists already)
```

> PrestaShop/kanbanbot :

Reviewer: ```suggestion
`PrestaShop/kanbanbot`:
```

- src/PullRequest/Domain/Aggregate/PullRequest/PullRequest.php
- src/PullRequest/Domain/Aggregate/PullRequest/PullRequestDescription.php
- tests/Shared/Infrastructure/Provider/TranslationsCatalogProviderTest.php
- tests/PullRequest/Application/CommandHandler/CheckTableDescriptionCommandHandlerTest.php

> The following repo should be updated by the QA autom team but anyway here are the files we updated.

Reviewer: ```suggestion
The QA Automation Team should update the following repositories:
```

> PrestaShop/ga.tests.ui.pr :

Reviewer: ```suggestion
`PrestaShop/ga.tests.ui.pr`:
```

---
title: Create a new version branch
weight: 20
aliases:
  - /maintainers-guide/releasing-prestashop/create-version-branch/
---

# Create a new branch for a new major/minor version.

## 1) create the new branch and push it to the main repo

## 2) open PRs based on the new branch and modify those files :

There are multiples repo to update.

I'll list the file I modified for the creation of the 9.0.x branch as well as the corresponding pull request.

Changed files :

PrestaShop/PrestaShop :  https://github.com/PrestaShop/PrestaShop/pull/37188

- .github/workflows/cron_js_routing.yml
- .github/workflows/cron_nightly_build.yml
- .github/workflows/cron_nightly_tests_9.0.x_mariadb.yml ( created )
- .github/workflows/cron_nightly_tests_9.0.x_mysql.yml ( created )
- .github/workflows/cron_nightly_tests_reports.yml
- .github/workflows/cron_nightly_tests_reusable.yml
- .github/workflows/cron_php_update_modules.yml
- .github/PULL_REQUEST_TEMPLATE.md

PrestaShop/test-scenarios : https://github.com/PrestaShop/test-scenarios/pull/558/files

- .github/workflows/gh-pages.yml
- config.json

> - src/content/scenarios/known-bugs/core/9.0.x.md ( created )
> - src/content/scenarios/known-bugs/_index.md

Reviewer 2: ```suggestion
- src/content/scenarios/known-bugs/core/9.0.x.md ( created )
- src/content/scenarios/known-bugs/_index.md
```

Don't modify these files, they will be created by Github Actions
